src/melee.cpp

```diff
@@ -539,7 +539,8 @@ bool Character::melee_attack_abstract( Creature &t, bool allow_special,
         add_msg( m_bad, _( "This weapon is too unwieldy to attack with!" ) );
         return false;
     }
-    int move_cost = attack_speed( *cur_weapon );
+    int move_cost = attack_speed( cur_weapon->has_flag( flag_UNARMED_WEAPON ) ? 
```

Comment:
Should this be used (instead of `attack_speed ( *cur_weapon )`) in the above check for whether a weapon is too unwieldy to attack with?

Reply:
That would make perfect sense, otherwise the calculation would be looking at the wrong thing. Will move move_cost above the check and use the move_cost as the value being checked against instead. No need to recalculate attack speed